### PR TITLE
Remove unused coordinator dynamic configs

### PR DESCRIPTION
**Pull request by kfaraz**
### Changes

- Remove unused coordinator dynamic configs `mergeBytesLimit` and `mergeSegmentsLimit`

### CODE REVIEW

File: server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java [hunk 1]
```diff
@@ -127,8 +123,6 @@ public CoordinatorDynamicConfig(
   {
     this.markSegmentAsUnusedDelayMillis =
         markSegmentAsUnusedDelayMillis;
-    this.mergeBytesLimit = mergeBytesLimit;
```
Comment (abhishekrb19) [inline server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java:130]: Curious - were these configs never used, or was their actual usage removed at some point, leaving just the configs in the code?  

Also, I see these configs are referenced in a few places like the docs and web-console: https://github.com/apache/druid/blob/master/docs/configuration/index.md#dynamic-configuration, https://github.com/apache/druid/blob/master/docs/api-reference/dynamic-configuration-api.md, https://github.com/apache/druid/blob/master/web-console/src/druid-models/coordinator-dynamic-config/coordinator-dynamic-config.tsx#L33

**Review by vogievetsky — APPROVED** [review]
Web console part looks good

**Review by abhishekrb19 — APPROVED** [review]
Thanks!

**Comment by kfaraz** [comment]
Thanks for the reviews, @abhishekrb19 , @vogievetsky !

**Comment by kfaraz** [comment]
Merging as failure is unrelated.

**Pull request merged**